store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -351,17 +387,22 @@ protected boolean isColumnQualifierBytesValid(final byte[] bytes, final int numP
 
     @Override
     public long buildTimestamp(final String group, final Properties properties) {
-        Long timestamp = null;
-        if (null != timestampProperty) {
-            timestamp = (Long) properties.get(timestampProperty);
+        Long timestamp;
+
+        // Allow override via property
+        if (timestampProperty != null) {
+            return (Long) properties.get(timestampProperty);
         }
 
-        if (null == timestamp) {
-            if (aggregatedGroups.contains(group)) {
-                timestamp = AccumuloStoreConstants.DEFAULT_TIMESTAMP;
-            } else {
-                timestamp = LongUtil.getTimeBasedRandom();
+        // Check if aggregating to see what timestamp we should apply
+        if (aggregatedGroups.contains(group)) {
+            timestamp = AccumuloStoreConstants.DEFAULT_TIMESTAMP;
```

Comment:
Perhaps this could be in an else after the `if (timeSensitiveAggregatedGroups.contains(group)) {` just so it isn't set twice. I promise I am not usually so nitpicky! Just really conscious about performance / making the logic flow as small as possible as this code will be run a lot.

Reply:
I think counter intuitively it can be more expensive to have more branches due to CPU branch prediction, if this assignment was running a method it would be bit more clear cut but it is a static variable which means this is very cheap to do. I think it will depend more on what hardware its ran on but ultimately the time complexity is the same either way so differences between would be negligible.

Reply:
My thinking was branch prediction wouldn't come into this as the if happens no matter what, and the else doesn't add anything on top of that. Might be completely wrong though I am no expert on this.